dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -155,11 +175,38 @@ private long getBackoffNanoseconds() {
 
         ListenableFuture<Response> failure(Throwable throwable) {
             if (++failures <= maxRetries) {
-                return retry(throwable);
+                if (shouldAttemptToRetry(throwable)) {
+                    return retry(throwable);
+                } else if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Not attempting to retry failure",
+                            SafeArg.of("serviceName", endpoint.serviceName()),
+                            SafeArg.of("endpoint", endpoint.endpointName()),
+                            throwable);
+                }
             }
             return Futures.immediateFailedFuture(throwable);
         }
 
+        private boolean shouldAttemptToRetry(Throwable throwable) {
+            if (retryOnSocketException == ClientConfiguration.RetryOnSocketException.DANGEROUS_DISABLED
+                    // IOException rather than SocketException Matches CJR for safety,
+                    // see RemotingOkHttpCall.should
+                    && throwable instanceof IOException) {
+                return false;
+            }
+            if (retryOnTimeout == ClientConfiguration.RetryOnTimeout.DISABLED) {
+                if (throwable instanceof SocketTimeoutException) {
+                    // non-connect timeouts should not be retried
+                    SocketTimeoutException socketTimeout = (SocketTimeoutException) throwable;
+                    return socketTimeout.getMessage() != null
+                            // String matches CJR RemotingOkHttpCall.shouldRetry
+                            && socketTimeout.getMessage().contains("connect timed out");
+                }
+            }
+            return true;
```

Comment:
one of these was implemented purely as a hack for alta, and if possible I'd kinda like to just not support it from the beginning to minimize complexity (as I think they can make single-uri channels to achieve their use-case).

Reply:
I've pulled it from this PR and added a check so we throw eagerly if the pref is enabled.

Reply:
Do you still want to keep the socketexception normalizing thing?

Reply:
Might as well get rid of it since ConnectTimeoutException isn't a socket exception